### DISCUSSION TITLE
Need information about Tauri

### DISCUSSION THREAD

**Discussion by luludelpeche** [tauri-apps/tauri#3546] [Q&A]
First of all, very nice project, I stumbled upon it and I have a few questions :

1) What does "alternative renderer" mean in the roadmap ?
2) Will it be able to access the mobile's native APIs when it becomes available (GPS, camera, etc.) ?
3) Can I create a JS button with user input, and call some RUST code to do a "heavy" calculation and return the result to the screen ?
4) When creating a Tauri project, I notice that several templates are offered to me (view, React, Solid, etc.), is there a way not to take a particular template and not do only simple HTML, CSS, JS ?
5) Could you use it for embedded systems ?
6) I made a donation, but I do not receive the PDF, I will see it again when the book is finished ?

Thanks !

**Comment by FabianLars** [comment]
1) Mainly (Open)GL, for example we had support for [egui](https://github.com/wusyong/egui), but had to remove it for v1 (will be re-added after v1 stable is released)
2) built-in tauri interfaces (aka "convenience functions") for this will be limited at first, but you will be able to do everything you want in a  DIY way (similar to how you have to use the windows-rs crate for really specific Windows interactions.
3) https://tauri.studio/docs/guides/command/
4) There is a vanilla js template. Alternatively you can create your frontend project first and then use `tauri init` -> https://tauri.studio/docs/getting-started/beginning-tutorial#1-add-tauri-to-existing-project
5) Well you _can_ (depending on the actual target), but a) it won't be easy to build for it and b) you probably don't even want to do it as browser engines (webviews) are generally just too heavy for embedded systems (again, depedning on the actual target).
6) You'll get the book/PDF once it's done & published, sorry for the delay and unclear communication.

**Reply by ITwrx** [reply]
why is there a link to opencollective for the book, but i don't see any book on opencollective? is it still not available? If not, i think this link should be removed and just have a message that says it's being worked on and solicit donations, in general. Unless i'm just missing the link, this is unnecessarily frustrating.
Also, i thought it would be one of these books where you get PDFs each release as the book is written, and it is expected that these pre-release versions are not perfect and may change each release. IIRC, I think the docs make it sound like that too.
